acceptance-tests/src/acceptance-test/java/tech/pegasys/teku/test/acceptance/VoluntaryExitAcceptanceTest.java

```diff
@@ -0,0 +1,145 @@
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveOutputStream;
+import org.apache.commons.compress.utils.IOUtils;
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.test.acceptance.dsl.AcceptanceTestBase;
+import tech.pegasys.teku.test.acceptance.dsl.BesuNode;
+import tech.pegasys.teku.test.acceptance.dsl.TekuDepositSender;
+import tech.pegasys.teku.test.acceptance.dsl.TekuNode;
+import tech.pegasys.teku.test.acceptance.dsl.TekuValidatorNode;
+import tech.pegasys.teku.test.acceptance.dsl.TekuVoluntaryExit;
+import tech.pegasys.teku.test.acceptance.dsl.tools.deposits.ValidatorKeyGenerator;
+import tech.pegasys.teku.test.acceptance.dsl.tools.deposits.ValidatorKeystoreGenerator;
+
+public class VoluntaryExitAcceptanceTest extends AcceptanceTestBase {
+
+  @Test
+  void test() throws Exception {
+    final BesuNode eth1Node = createBesuNode();
+    eth1Node.start();
+
+    int numberOfValidators = 4;
+    final TekuDepositSender depositSender = createTekuDepositSender();
+    final List<ValidatorKeyGenerator.ValidatorKeys> validatorKeys =
+        depositSender.generateValidatorKeys(numberOfValidators);
+    depositSender.sendValidatorDeposits(eth1Node, validatorKeys, MAX_EFFECTIVE_BALANCE);
+
+    final String validatorKeysPassword = "validatorsKeysPassword";
+    final Path validatorInfoDirectoryPath = Path.of("./validatorInfo");
+    final String keysDirectory = "keys";
+    final String passwordsDirectory = "passwords";
+    final Path keysOutputPath = validatorInfoDirectoryPath.resolve(keysDirectory);
+    final Path passwordsOutputPath = validatorInfoDirectoryPath.resolve(passwordsDirectory);
+    final ValidatorKeystoreGenerator keystoreGenerator =
+        new ValidatorKeystoreGenerator(
+            validatorKeysPassword, keysOutputPath, passwordsOutputPath, (__) -> {});
+
+    // create temporary tar file which will be copied to docker containers
+    File validatorInfoTar = File.createTempFile("validatorInfo", ".tar");
+    validatorInfoTar.deleteOnExit();
+
+    // create keystores using the validator keys generated by deposit sender
+    keystoreGenerator.generateKeystoreAndPasswordFiles(
+        validatorKeys.stream()
+            .map(ValidatorKeyGenerator.ValidatorKeys::getValidatorKey)
+            .collect(Collectors.toList()));
+
+    // copy keystores directory to tar file and delete the now redundant directory
+    copyDirectoryToTarFile(validatorInfoDirectoryPath, validatorInfoTar.toPath());
+    FileUtils.deleteDirectory(validatorInfoDirectoryPath.toFile());
```

Comment:
We should push all this logic into the DSL rather than doing it directly in the test.  The AT itself should be just high level steps like
```
keys = depositSender.sendValidatorDeposits(..)
createValidatorNode(beaconNode, keys)
sendVoluntaryExit(beaconNode, keys)
```

Reply:
Yep. Will do.